src/main/java/cashleh/ExpenseStatement.java

```diff
@@ -14,16 +17,35 @@ public ExpenseStatement(Expense... expenses) {
     public void addExpense(Expense expenseToAdd) {
         expenseStatement.add(expenseToAdd);
     }
-    public void deleteExpense(int expenseIndexToDelete) {
-        expenseStatement.remove(expenseIndexToDelete);
+    public void deleteExpense(int expenseIndex) throws CashLehMissingTransactionException {
+        try {
+            expenseStatement.remove(expenseIndex);
+        } catch(IndexOutOfBoundsException e) {
+            throw new CashLehMissingTransactionException();
+        }
     }
-    public Expense getExpense(int expenseIndex) {
-        return expenseStatement.get(expenseIndex);
+    public Expense getExpense(int expenseIndex) throws CashLehMissingTransactionException {
+        try {
+            return expenseStatement.get(expenseIndex);
+        } catch (IndexOutOfBoundsException e) {
+            throw new CashLehMissingTransactionException();
+        }
     }
     public int getNumberOfExpenses() {
         return expenseStatement.size();
     }
 
+    public void getExpenses() {
```

Comment:
I think that this method should be renamed to printExpenses as it returns voic and prints text instead